### PR TITLE
OPS-772 barcode generation

### PR DESCRIPTION
**Pull request by photomattmills**

### CODE REVIEW

File: lib/image_encoder.ex [hunk 1]
```diff
@@ -0,0 +1,93 @@
+defmodule PDF417.PNGEncoder do
```
Comment (tielur) [inline lib/image_encoder.ex]: Can we rename this file to `png_encoder.ex`? I was looking for this and took me a hot minute to find it 😆

**Review by tielur — APPROVED** [review]
Looks good 👍 I think adding credo would be nice and maybe some moduledocs for things like the high level and low level encoder.

File: lib/image_encoder.ex (hunk 1 above)
Comment (photomattmills) [reply lib/image_encoder.ex]: weird, I thought I did that. that's what I get for working in the wee hours of the morning.

**Comment by tielur** [comment]
All of the files in lib besides `pdf417.ex` should probably be put in a **pdf417/** folder to follow the namespace but it doesn't hurt anything to leave it how it is

File: lib/pdf417.ex [hunk 2]
```diff
@@ -1,19 +1,43 @@
...
+  * `:security_level` - Defaults to 3, which is sufficient for up to 160 data codewords (about 320 characters of text). Every 320 characters, add one level here, up to level 8.
+  """
+  def encode(message, options \\ %{}) do
     options =
       @default_options
-      |> Map.merge(opts)
+      |> Map.merge(options)
       |> Map.merge(%{message: message})
 
     HighLevelEncoder.encode(options)
```
Comment (tielur) [inline lib/pdf417.ex:23]:
```suggestion
    options
    |> HighLevelEncoder.encode()
```

Comment (tielur) [reply lib/pdf417.ex:23]: Just following the "start pipes with raw values" credo rule

**Pull request merged**